reportek/core/models/reporting.py

```diff
@@ -213,35 +213,62 @@ def validate_filename(value):
         )
 
 
-class EnvelopeOriginalFile(models.Model):
+class EnvelopeFileQuerySet(models.QuerySet):
+
+    def delete(self):
+        for obj in self:
+            try:
+                os.remove(obj.file.path)
+                debug(f'Deleted disk file {obj.file.path}')
+            except FileNotFoundError:
+                error(f'Could not delete envelope file from disk (not found): '
+                      f'{obj.file.path}')
+        super().delete()
+
+
+class EnvelopeFileCommon(models.Model):
```

Comment:
Do you think it would make sense to call this `BaseEnvelopeFile`, in keeping with the nomenclature elsewhere in the project (e.g. `BaseWorkflow`,`BasePermission`, `BaseEnvelopeFilePermissions`, etc.) ?

Reply:
Yes it would.